### PR TITLE
Issue #841: Skupper images run as root

### PR DESCRIPTION
**Pull request by ajssmith**
Setup user for container and define PodSecurityContext for deployments

### CODE REVIEW

**Review by nluaces — APPROVED** [review]

File: Dockerfile.config-sync [hunk 1]
```diff
@@ -9,9 +9,14 @@ RUN go mod download
...
 RUN make
 
-
 FROM registry.access.redhat.com/ubi8-minimal
 
+# Create user and group and switch to user's context
+RUN microdnf -y install shadow-utils \
+&& microdnf clean all
+RUN useradd --uid 10000 runner
+USER 10000
```
[the remark this answers is not in the document]
Comment (ajssmith) [reply Dockerfile.config-sync:18]: A nice feature of OpenShift. On other platforms, containers are running as root. Hardening best practices suggest running as non-root user.

**Review by fgiorgetti — COMMENTED** [review]
It seems to work for me (locally), but it would be nice to test as well against
multiple clusters and versions to make sure it is OK.

@granzoto I believe you have done similar evaluation recently.
Could you validate this one as well?

**Review by grs — APPROVED** [review]

File: cmd/site-controller/deploy-watch-all-ns.yaml [hunk 2]
```diff
@@ -135,3 +135,11 @@ spec:
...
       - name: site-controller
         image: quay.io/skupper/site-controller:master
+        securityContext:
+          capabilities:
+            drop:
+            - ALL
+          runAsNonRoot: true
+          allowPrivilegeEscalation: false
+          seccompProfile:
+            type: RuntimeDefault
```
Comment (fgiorgetti) [inline cmd/site-controller/deploy-watch-all-ns.yaml:145]: The `seccompProfile` field does not work with OKD 3.11.
All other fields seem to be working just fine.

Do you think it is better to keep it as is and just drop a note
somewhere explaining to remove it if running on 3.11 ?

File: client/router_create.go [hunk 3]
```diff
@@ -1079,7 +1079,19 @@ sasldb_path: /tmp/skrouterd.sasldb
 			}
 		}
 	}
-	dep, err := kube.NewTransportDeployment(van, siteOwnerRef, cli.KubeClient)
+
+	runAsNonRoot := true
+	podSecurityContext := corev1.PodSecurityContext{
```
Comment (fgiorgetti) [inline client/router_create.go:1084]: One comment/question here. The `corev1.PodSecurityContext` type is not
the same one we are setting at the YAML files, which uses `v1.SecurityContext` type,
and that one is defined at the container level.

Should the same fields be set on both places?

**Comment by fgiorgetti** [comment]
Just some more context:
https://sdk.operatorframework.io/docs/best-practices/pod-security-standards/#how-should-i-configure-my-operators-and-operands-to-comply-with-the-criteria

File: cmd/site-controller/deploy-watch-all-ns.yaml (hunk 2 above)
Comment (ajssmith) [reply cmd/site-controller/deploy-watch-all-ns.yaml:145]: Lean towards keeping as is and will add a comment to indicate version considerations

File: client/router_create.go (hunk 3 above)
Comment (ajssmith) [reply client/router_create.go:1084]: Agree and will update

**Review by granzoto — APPROVED** [review]
LGTM

**Pull request merged**